http/src/main/java/io/micronaut/http/MediaType.java

```diff
@@ -850,7 +886,11 @@ public static MediaType forFilename(String filename) {
                 String[] tokens = formattedLine.split("\\|");
                 for (int i = 1; i < tokens.length; i++) {
                     String fileExtension = tokens[i].toLowerCase(Locale.ENGLISH);
-                    result.put(fileExtension, tokens[0]);
+                    if (keyedByExtension) {
```

Comment:
Have to rethink this since there can be multiple extensions per media type, can't just choose a random one.